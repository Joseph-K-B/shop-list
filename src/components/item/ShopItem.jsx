import ItemControls from "../controls/ItemControls";

function ShopItem({item, onChange}) {

  return(
    <>
      <h3>{item.name}</h3>
      <ItemControls item={item} onChange={onChange}/>     
    </>
  )
}

export default ShopItem;