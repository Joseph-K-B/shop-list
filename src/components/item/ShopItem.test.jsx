import React from 'react';
import { screen, render } from '@testing-library/react';
import ShopItem from './ShopItem';

it('renders item name and controls', () => {
    const {container} = render(
        <ShopItem item={{id: 1, name: 'test', done: false}} />
    );

    const item = screen.getByLabelText(/list-item/i)

    expect(item).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});