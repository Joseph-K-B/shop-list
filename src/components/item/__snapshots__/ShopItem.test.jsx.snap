// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders item name and controls 1`] = `
<div>
  <section
    aria-label="list-item"
  >
    <div
      class="itemInfo"
    >
      <h3>
        test
      </h3>
    </div>
    <div
      class="itemControls"
    >
      <div>
        <input
          class="check"
          type="checkbox"
          value="false"
        />
      </div>
      <div>
        <button
          aria-label="open delete form"
          type="button"
        >
          Remove
        </button>
      </div>
      <button
        aria-label="open update form"
      >
        Edit
      </button>
    </div>
  </section>
</div>
`;
